models/User/Service.php

```diff
@@ -29,13 +31,17 @@ class Service
      *
      * @return string|null
      */
-    public static function getClassNameForType($type)
+    public static function getClassNameForType($type): ?string
     {
         switch ($type) {
-            case 'user': return '\\Pimcore\\Model\\User';
-            case 'userfolder': return '\\Pimcore\\Model\\User\\Folder';
-            case 'role': return '\\Pimcore\\Model\\User\\Role';
-            case 'rolefolder': return '\\Pimcore\\Model\\User\\Role\\Folder';
+            case 'user':
+                return User::class;
+            case 'userfolder':
+                return User\Folder::class;
+            case 'role':
+                return User\Role::class;
+            case 'rolefolder':
+                return User\Role\Folder::class;
             default:
                 return null;
         }
```

Comment:
This is a great use-case for PHP 8's new [`match` expressions](https://www.php.net/manual/en/control-structures.match.php):

```suggestion
        return match ($type) {
            'user' => User::class,
            'userfolder' => User\Folder::class,
            'role' => User\Role::class,
            'rolefolder' => User\Role\Folder::class,
            default => null,
        }
```

Reply:
The suggestion that GitHub created is somehow broken O.o

Reply:
@jdreesen I can't commit it :( I changed it here: https://github.com/pimcore/pimcore/pull/12026/commits/2d58443f5c8a1a01ca594234df6f5a0efeefad5c